src/external/bitcoin.rs

```diff
@@ -74,51 +75,52 @@ pub async fn get_transactions(address: String) -> Transactions {
     };
     let transactions: Transactions = reqwest::get(&request_url)
         .await
-        .expect("Failed to send request")
+        .context("Failed to send request")?
         .json()
         .await
-        .expect("Failed to read response");
+        .context("Failed to read response")?;
 
-    transactions
+    Ok(transactions)
 }
 
 impl Txid {
-    pub async fn get_first_transaction(transactions: Transactions) -> Self {
-        transactions.last().unwrap().clone()
+    pub async fn get_first_transaction(transactions: Transactions) -> Result<Self> {
+        transactions.last().cloned().context("No transactions found")
     }
 }
 
-pub async fn get_last_block_height() -> u64 {
+pub async fn get_last_block_height() -> Result<u64> {
     let request_url = match *USERNETWORK {
         Network::Bitcoin => "https://blockstream.info/api/blocks/tip/height",
         _ => "https://blockstream.info/testnet/api/blocks/tip/height",
     };
     let height: u64 = reqwest::get(request_url)
         .await
-        .expect("Failed to send request")
+        .context("Failed to send request")?
         .json()
         .await
-        .expect("Failed to read response");
+        .context("Failed to read response")?;
 
-    height
+    Ok(height)
 }
 
-pub async fn check_if_paid(address: String, amount: u64) -> (bool, u64) {
+pub async fn check_if_paid(address: String, amount: u64) -> Result<(bool, u64)> {
     //todo check what net we used
-    let info_about_address = AddressInfo::get_address_info(address.clone()).await;
+    let info_about_address = AddressInfo::get_address_info(address.clone()).await?;
     let received_summ = info_about_address.chain_stats.funded_txo_sum;
     let spent_summ = info_about_address.chain_stats.spent_txo_sum;
     let received_summ_mempool = info_about_address.mempool_stats.funded_txo_sum;
     let spent_summ_mempool = info_about_address.mempool_stats.spent_txo_sum;
     if amount.eq(&(received_summ + spent_summ + received_summ_mempool + spent_summ_mempool)) {
-        (true, received_summ)
+        Ok((true, received_summ))
     } else {
-        (false, 0)
+        Ok((false, 0))
     }
 }
 
-pub fn get_address_to_pay(bill: BitcreditBill) -> String {
-    let public_key_bill = bitcoin::PublicKey::from_str(&bill.public_key).unwrap();
+pub fn get_address_to_pay(bill: BitcreditBill) -> Result<String> {
```

Comment:
There are compilation errors in the code calling this (bill_service). If you add error-handling, you have to add it on both sides (caller and callee)